### PR TITLE
[AG-612] algebrav1 fee approximation

### PR DESCRIPTION
**Pull request by it4rb**

### CODE REVIEW

File: pkg/source/algebrav1/pool_tracker.go [hunk 1]
```diff
@@ -188,9 +190,261 @@ func (d *PoolTracker) fetchRPCData(ctx context.Context, p entity.Pool) (FetchRPC
...
 	return res, err
 }
 
+func (d *PoolTracker) approximateFee(ctx context.Context, poolAddress, dataStorageOperator string, state *GlobalState, currentLiquidity *big.Int) error {
+	// fee approximation: assume that the swap will be soon after this
+	blockTimestamp := uint32(time.Now().Unix())
+	yesterday := blockTimestamp - WINDOW
+	timepoints, err := d.getPoolTimepoints(ctx, state.TimepointIndex, poolAddress, yesterday)
+	if err != nil {
+		return err
```
Comment (piavgh) [inline pkg/source/algebrav1/pool_tracker.go:207]: we should add an error log here to detect the problem easier

File: pkg/source/algebrav1/pool_tracker.go [hunk 2]
```diff
@@ -188,9 +190,261 @@ func (d *PoolTracker) fetchRPCData(ctx context.Context, p entity.Pool) (FetchRPC
...
+	}
+
+	if timepoints == nil {
+		// not initialized pool has been locked already, but set here just for sure
+		state.Unlocked = false
```
Comment (piavgh) [inline pkg/source/algebrav1/pool_tracker.go:212]: it's better to add an info log here (to answer the question: why the route does not go through this pool?)

Comment (it4rb) [reply pkg/source/algebrav1/pool_tracker.go:222]: I've added log in the parent function, to catch the case when it's locked at the SC

File: pkg/source/algebrav1/data_storage.go [hunk 3]
```diff
@@ -0,0 +1,482 @@
...
+		BlockTimestamp:                0,
+		TickCumulative:                0,
+		SecondsPerLiquidityCumulative: new(big.Int),
+		VolatilityCumulative:          new(big.Int),
+		AverageTick:                   0,
+		VolumePerLiquidityCumulative:  new(big.Int),
+	}
+}
+func (s *TimepointStorage) Set(index uint16, v Timepoint) {
+	s.updates[index] = v
```
Comment (piavgh) [inline pkg/source/algebrav1/data_storage.go:40]: why don't we need to set the s.data here?

Comment (it4rb) [reply pkg/source/algebrav1/data_storage.go:40]: `s.data` is the original data fetched from SC
`s.updates` will contains the written data during simulation or fee approximation, and will be reverted after each calculation (this code is originally for simulator, but then got moved to tracker for fee approximation)
In the Get function we'll check from both `s.data` and `s.updates`

**Review by piavgh — APPROVED** [review]

**Pull request merged**